api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -0,0 +1,265 @@
+    }
+  }
+
+  // Converts the hex numbers stored as strings into bytes stored in a buffer.
+  static void GenerateHexFromString(nostd::string_view string, int bytes, uint8_t *buf)
+  {
+    const char *str_id = string.begin();
+    for (int i = 0; i < bytes; i++)
+    {
+      int tmp = HexToInt(str_id[i]);
+      if (tmp < 0)
+      {
+        for (int j = 0; j < bytes / 2; j++)
+        {
+          buf[j] = 0;
+        }
+        return;
+      }
+      if (i % 2 == 0)
+      {
+        buf[i / 2] = tmp * 16;
+      }
+      else
+      {
+        buf[i / 2] += tmp;
+      }
+    }
+  }
+
+  static TraceId GenerateTraceIdFromString(nostd::string_view trace_id)
+  {
+    uint8_t buf[kHeaderElementLengths[1] / 2];
+    uint8_t *b_ptr = buf;
+    GenerateHexFromString(trace_id, kHeaderElementLengths[1], b_ptr);
+    return TraceId(buf);
+  }
+
+  static SpanId GenerateSpanIdFromString(nostd::string_view span_id)
+  {
+    uint8_t buf[kHeaderElementLengths[2] / 2];
+    uint8_t *b_ptr = buf;
+    GenerateHexFromString(span_id, kHeaderElementLengths[2], b_ptr);
+    return SpanId(buf);
+  }
+
+  static TraceFlags GenerateTraceFlagsFromString(nostd::string_view trace_flags)
+  {
+    if (trace_flags.length() > 2)
+    {
+      return TraceFlags(0);  // check for invalid length of flags
+    }
+    int tmp1 = HexToInt(trace_flags[0]);
+    int tmp2 = HexToInt(trace_flags[1]);
+    if (tmp1 < 0 || tmp2 < 0)
+      return TraceFlags(0);  // check for invalid char
+    uint8_t buf = tmp1 * 16 + tmp2;
+    return TraceFlags(buf);
+  }
+
+private:
+  // Converts a single character to a corresponding integer (e.g. '1' to 1), return -1
+  // if the character is not a valid number in hex.
+  static uint8_t HexToInt(char c)
+  {
+    if (c >= '0' && c <= '9')
+    {
+      return (int)(c - '0');
+    }
+    else if (c >= 'a' && c <= 'f')
+    {
+      return (int)(c - 'a' + 10);
+    }
+    else if (c >= 'A' && c <= 'F')
+    {
+      return (int)(c - 'A' + 10);
+    }
+    else
+    {
+      return -1;
+    }
+  }
+
+  static void InjectTraceParent(const SpanContext &span_context, T &carrier, Setter setter)
+  {
+    char trace_id[32];
+    TraceId(span_context.trace_id()).ToLowerBase16(trace_id);
+    char span_id[16];
+    SpanId(span_context.span_id()).ToLowerBase16(span_id);
+    char trace_flags[2];
+    TraceFlags(span_context.trace_flags()).ToLowerBase16(trace_flags);
+    // Note: This is only temporary replacement for appendable string
+    std::string hex_string = "00-";
+    for (int i = 0; i < 32; i++)
+    {
+      hex_string.push_back(trace_id[i]);
+    }
+    hex_string.push_back('-');
+    for (int i = 0; i < 16; i++)
+    {
+      hex_string.push_back(span_id[i]);
+    }
+    hex_string.push_back('-');
+    for (int i = 0; i < 2; i++)
+    {
+      hex_string.push_back(trace_flags[i]);
+    }
+    setter(carrier, kTraceParent, hex_string);
+  }
+
+  static void InjectImpl(Setter setter, T &carrier, const SpanContext &span_context)
```

Comment:
I think all is cool except for the GenerateHexFromString, the rest are useful because they are useful functions outside of this class as well and they also need tests to make sure they are correct, although I have changed the GenerateHexFromString to private.